node/src/main/scala/com/wavesplatform/state/diffs/CommonValidation.scala

```diff
@@ -95,6 +100,13 @@ object CommonValidation {
 
           for {
             address <- blockchain.resolveAlias(citx.dApp)
+            _ <- citx.payments
+              .collectFirstSome {
+                case Payment(_, IssuedAsset(id)) => InvokeDiffsCommon.checkAsset(blockchain, id).swap.toOption
+                case Payment(_, Waves)           => None
+              }
```

Comment:
Why just don't use `.collectFirst` which works the same way but with less code?
```suggestion
            _ <- citx.payments
              .collectFirst { 
                case Payment(_, IssuedAsset(id)) => InvokeDiffsCommon.checkAsset(blockchain, id).swap
              }
```

Reply:
Because it's not the same. 
`.collectFirstSome` allows here to collect error from `Left`, avoiding `Either` structure.